src/netCDF4/_netCDF4.pyx

```diff
@@ -2246,7 +2255,7 @@ strings.
                     ierr = nc_open(path, NC_NOWRITE | NC_SHARE, &grpid)
                 else:
                     ierr = nc_open(path, NC_NOWRITE, &grpid)
-        elif mode == 'r+' or mode == 'a':
+        elif mode == 'r+' or mode == 'a' and os.path.exists(filename):
```

Comment:
`and` and `or` does not have the same precedence, this is read as `mode == r+ or (mode == a and os.path.exists(filename)`, didn't you mean: `mode in ('r+', 'a') and os.path.exists(filename)`? Same problem further down.

Reply:
good catch - will fix